src/Feature/Theme/Feature_Theme.re

```diff
@@ -0,0 +1,101 @@
+open Oni_Core;
+open Revery;
+
+module Log = (val Oni_Core.Log.withNamespace("Oni2.Feature.Theme"));
+
+module Colors = GlobalColors;
+
+type model = {
+  defaults: ColorTheme.Defaults.t,
+  theme: ColorTheme.t,
+};
+
+let defaults =
+  [
+    Colors.Editor.defaults,
+    Colors.List.defaults,
+    Colors.Oni.defaults,
+    Colors.SideBar.defaults,
+    Colors.Tab.defaults,
+    Colors.remaining,
+  ]
+  |> List.map(ColorTheme.Defaults.fromList)
+  |> ColorTheme.Defaults.unionMany;
+
+let initial = contributions => {
+  defaults: ColorTheme.Defaults.unionMany([defaults, ...contributions]),
+  theme: ColorTheme.{variant: Dark, colors: ColorTheme.Colors.empty},
+};
```

Comment:
The contributions are passed in here and combined with the local defaults to be stored in the state. It could also be stored outside the state, but would then have to be passed in alongside the model where needed.